hack/generator/pkg/astmodel/resource_registration_file.go

```diff
@@ -0,0 +1,327 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+// TODO: I am not sure that it makes sense to have this file in astmodel at all... maybe just in code_generator?
```

Comment:
Thoughts?

Reply:
Reading it I agree with the comment, this should probably be in code_generator - it's building some AST parts, but it's not really part of the model.

Reply:
I agree - moved it